packages/client/src/generation/utils/buildInjectableEdgeEnv.ts

```diff
@@ -1,4 +1,4 @@
-import type { InternalDatasource } from '../../runtime/utils/printDatasources'
```

Comment:
Did you check that the types were different?
See 
```
export interface DataSource {
  name: string
  provider: ConnectorType
  activeProvider: ConnectorType
  url: EnvValue
  // below is different from InternalDatasource
  directUrl?: EnvValue
  schemas: string[] | []
}

export interface InternalDatasource {
  name: string
  provider: ConnectorType
  activeProvider: ConnectorType
  url: EnvValue
  // below is different from Datasource
  config: any
}
```

Reply:
Yes I know they were different and caused conflicts in type-checking in the code base. The original type that we get is always of type `DataSource` but somehow along the way it was replaced with `InternalDatasource` without adding or using these extra properties, essentially casting `DataSource` to `InternalDatasource`. I trust `DataSource` is the correct one because that is the type given by the generator:
![image](https://github.com/prisma/prisma/assets/18401805/1c5bd3bc-634b-413b-92a9-799212d14eb8)